layouts/partials/navbar.html

```diff
@@ -56,6 +56,12 @@
             {{ if $external }}target="_blank" rel="noreferer"{{ end }}
             class="hx-text-sm contrast-more:hx-text-gray-700 contrast-more:dark:hx-text-gray-100 hx-relative -hx-ml-2 hx-hidden hx-whitespace-nowrap hx-p-2 md:hx-inline-block {{ $activeClass }}"
           >
+            {{ with .Params.hero_icon -}}
```

Comment:
```suggestion
            {{ with .Params.icon -}}
```

I'd prefer `icon` over `hero_icon`

Reply:
Ah, but `.icon` is what sets it as icon _only_, I believe. That's why I went with `hero_icon`. E.g.:

```yaml
  - name: GitHub
    weight: 5
    url: "https://github.com/dogweather/forkful"
    params:
      icon: github
```

Reply:
I don't know how much you want to refactor this. Personally, I do "making illegal states impossible". We'd use `type` for these as well (it's already used for `type: search`. That could look like this:

```yaml
- name: github
  params:
    type: icon
    icon: github
- name: forum
  params:
    type: url
    icon: chat-alt-2
```

Reply:
I like the idea of using an extra `type` there, let's consider these cases

```yaml
# Icon only, for backward compatibility
- name: github
  url: https://github.com/
  params:
    icon: github

# Link with icon, must set `type: link` and `icon`
- name: forum
  params:
    type: link
    icon: chat-alt-2

# Search bar only
- name: Search
  params:
    type: search
    icon: chat-alt-2 # undefined behavior, currently ignored
```

In the future, we may extend the `type` to other widgets, for example, light/dark/system theme toggle switch:
```yaml
- name: Theme toggle
  params:
    type: theme-toggle
```